### PR TITLE
Rust analyzer

### PR DESCRIPTION
**Pull request by scruffystuffs**
Rust analyzer

check https://github.com/fossas/spectrometer/blob/a1b57a99721af76261d6165456cd84580d3ff34c/docs/strategies/rust.md for details on the abstract analysis process.

Please note that `Strategy.Cargo (Package, PackageDependency)` are added for future use, when determining licenses.  Since we don't hold that info yet, we are not using that code.

### CODE REVIEW

**Comment by scruffystuffs** [comment]
Question: should the top level analyzed project be considered a direct dependency?  Current iteration is doing this, but that doesn't make sense to me.  I'll confirm before merging.

**Comment by scruffystuffs** [comment]
Pretty-printed output from a rust project of mine, found at https://github.com/scruffystuffs/jfmt.rs
Uploaded as text because github doesn't allow `.json` files (too scary).
[jfmt.json.txt](https://github.com/fossas/spectrometer/files/4618790/jfmt.json.txt)

**Comment by scruffystuffs** [comment]
> Question: should the top level analyzed project be considered a direct dependency? Current iteration is doing this, but that doesn't make sense to me. I'll confirm before merging.

Btw, the answer was yes, which is why #65 was created and merged.

File: src/Strategy/Cargo.hs [hunk 1]
```diff
@@ -0,0 +1,207 @@
+module Strategy.Cargo
+  ( discover
+  --
+  , CargoMetadata(..)
+  , NodeDependency(..)
+  , NodeDepKind(..)
+  -- , Package(..)
```
Comment (cnr) [inline src/Strategy/Cargo.hs:7]: [nit] we should kill the comments here and elsewhere

**Pull request merged**